### PR TITLE
docs: add new benchmark page with SIFT1M dataset

### PR DESCRIPTION
**Pull request by AnneYang720**
Goals:

This PR will add sift1m dataset to benchmark scripts according to https://github.com/jina-ai/docarray/issues/298.
The code will continue from the work of https://github.com/jina-ai/docarray/pull/514 , https://github.com/jina-ai/docarray/pull/301 and https://github.com/jina-ai/docarray/pull/310. 

Change PR https://github.com/jina-ai/docarray/pull/614 to jina branch.

- [x] adapt benchmark script to latest docarray
- [x] add sift1m dataset
- [x] reuse code in both random and dataset benchmarking
- [x] add code to generate benchmark recall lines 
- [x] update benchmarking results

### CODE REVIEW

File: scripts/benchmarking_utils.py
Comment (alaeddine-13) [inline scripts/benchmarking_utils.py:490]:
```suggestion
                da = DocumentArray(storage=storage, config={k: v for k, v in config.items() if k not in {'num_candidates', 'hnsw_ef'}})
```

File: scripts/benchmarking_utils.py
Comment (alaeddine-13) [inline scripts/benchmarking_utils.py:620]:
```suggestion
    return param_dict.get(storage, {}).get(param, param)
```

File: scripts/README.md
Comment (alaeddine-13) [inline scripts/README.md:34]: how do we select sift or random dataset ?

File: scripts/README.md
Comment (alaeddine-13) [inline scripts/README.md:22]: why is common needed here ?

Comment (AnneYang720) [reply scripts/README.md:22]: `matplotlib` is needed to generate the figure

File: scripts/README.md
Comment (AnneYang720) [reply scripts/README.md:38]: I added docs for random dataset now. But I think the scripts for random dataset should be removed in the future.

**Comment by bwanglzu** [comment]
i like our old benchmark (docarray benchmark), with colors, green indicates it's faster and red is slower. otherwise the table is not easy to read and compare

File: scripts/README.md
Comment (JoanFM) [reply scripts/README.md:22]: I would suggest adding all to the `benchmark` part

**Comment by JoanFM** [comment]
Can we have a `chart` with the `query by condition` comparison? It is not clear the charts what are they?

File: scripts/README.md
Comment (alexcg1) [inline scripts/README.md:25]: Is it SIFT1M or sift1m? Let's be consistent throughout the whole doc

File: scripts/README.md
Comment (alexcg1) [inline scripts/README.md:43]:
```suggestion
The results of the benchmarks are stored in the current directory:
- `benchmark-seconds-{storage}.csv`
- `benchmark-qps-{n}.csv` 
- `benchmark.png`, a figure of of `Recall@10/QPS` for all storage backends
```

What does the figure show?

File: scripts/README.md
Comment (alexcg1) [inline scripts/README.md:48]:
```suggestion
Current HNSW parameters are written in the function `get_configuration_storage_backends` of `benchmarking_utils.py`. Parameters include:
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: scripts/README.md
Comment (alaeddine-13) [reply scripts/README.md:22]: yes at least we shouold remove common

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:12]: I'm not sure whether the link from tensorflow is the official sift1m page I think the home page is http://corpus-texmex.irisa.fr/

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:58]: let's change None

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:164]: change None

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:91]: higher recall as the ef value increase implies that the configuration is correctly applied, although find by vector stays constant.
I doubt big network overhead here, I'm trying to confirm

File: scripts/README.md
Comment (alaeddine-13) [inline scripts/README.md:32]: btw when I tried python benchmarking.py --help it looks like the script is loading data, can you check this ?

**Comment by alaeddine-13** [comment]
let's also change `scroll_batch_size` to something more fair

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:352]:
```suggestion
AnnLite offers good speed in CRUD and vector search operations, but keep in mind that AnnLite is a library and does not follow a client-server design. This means that AnnLite operations do not include network overhead unlike other backends.

Moreover, AnnLite is a local monolithic package and does not offer scaling. However, you can still scale using [Jina's scaling features](https://docs.jina.ai/how-to/scale-out/).

```

**Review by NicholasDunham — CHANGES_REQUESTED** [review]

**Review by alaeddine-13 — CHANGES_REQUESTED** [review]

**Review by NicholasDunham — APPROVED** [review]

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [reply docs/advanced/document-store/benchmark.md:27]: and for the filter operation, let's say it uses backend side filtering when possible

Comment (alaeddine-13) [reply docs/advanced/document-store/benchmark.md:27]: in the text below, instead of saying:

> We use similar underlying search algorithms but different implementations**: In this benchmark we focus on setting only parameters `ef`, `ef_construct` and `max_connections` from HNSW

let's say

> We use similar underlying search algorithms but different implementations**: In this benchmark we try a set of parameters `ef`, `ef_construct` and `max_connections` from HNSW applied equally on all backends.

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:171]: > except for `Recall@10`

did you mean something else ?
because also high recall is good

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:336]: I think we should say here that we show the trade-off between quality and speed as you tune your parameters in each document store

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:342]: I'm referring to  the text below:

> [raise a PR or issue](https://github.com/jina-ai/docarray/issues/new/choose).
let's change this link to the new docarray repository

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:366]: this belongs to the removed chart, we should also remove this

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:370]:
```suggestion

Moreover, AnnLite is a local monolithic package and does not offer scaling. However, you can still scale using [Jina's scaling features](https://docs.jina.ai/how-to/scale-out/).
```

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:363]: we still don't talk about other backends. I think we can share the following insight:
* Redis offers great speed if your data fits in memory and you need an isolated server to store your data

**Review by alaeddine-13 — CHANGES_REQUESTED** [review]
let's wait until weaviate PR is merged and use the same batch_size for all: 64
 We also need to mention this fact in benchmark.md

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:21]: update qdrant version

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:457]:
```suggestion
#### Client/Server setup
```
Let's say here that in our benchmarks the client and server live on the same machine to minimize network overhead.
We need to say that although a real life scenario would be clients and server living  on different machines with potentially multiple clients in parallel, we choose to keep both in the same machine and have only 1 client process

**Review by alaeddine-13 — DISMISSED** [review]

File: docs/advanced/document-store/benchmark.md
Comment (alaeddine-13) [inline docs/advanced/document-store/benchmark.md:20]: update weaviate version

**Review by JoanFM — APPROVED** [review]

**Pull request merged**
